worldedit-core/src/main/java/com/sk89q/worldedit/regions/Polygonal2DRegion.java

```diff
@@ -197,18 +199,24 @@ public BlockVector3 getMaximumPoint() {
     }
 
     @Override
-    public int getArea() {
-        double area = 0;
+    public long getVolume() {
+        BigDecimal area = BigDecimal.ZERO;
         int i, j = points.size() - 1;
 
         for (i = 0; i < points.size(); ++i) {
-            area += (points.get(j).getBlockX() + points.get(i).getBlockX())
-                    * (points.get(j).getBlockZ() - points.get(i).getBlockZ());
+            area = area.add(
+                    BigDecimal.valueOf(points.get(j).getBlockX() + points.get(i).getBlockX())
+                            .multiply(BigDecimal.valueOf(points.get(j).getBlockZ() - points.get(i).getBlockZ()))
+            );

```

Comment:
This looks a little ugly -- perhaps pull out the X and Z parts to variables with names?